sdk/core/perf/src/base_test.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/perf/base_test.hpp"
+#include "azure/core/http/policies/policy.hpp"
+#include "azure/core/internal/http/pipeline.hpp"
+
+#include <functional>
+#include <string>
+#include <vector>
+
+using namespace Azure::Core::Http;
+using namespace Azure::Core::Http::Policies;
+using namespace Azure::Core;
+namespace {
+
+class ProxyPolicy final : public HttpPolicy {
+private:
+  Azure::Perf::BaseTest* m_testContext;
+
+public:
+  ProxyPolicy(Azure::Perf::BaseTest* proxyManager) : m_testContext(proxyManager) {}
+
+  // copy
+  ProxyPolicy(ProxyPolicy const& other) : ProxyPolicy{other.m_testContext} {}
+
+  // move
+  ProxyPolicy(ProxyPolicy&& other) : m_testContext{other.m_testContext} {}
+
+  std::unique_ptr<RawResponse> Send(
+      Request& request,
+      NextHttpPolicy nextPolicy,
+      Context const& context) const override
+  {
+    std::string const recordId(m_testContext->m_recordId);
+    if (recordId.empty())
+    {
+      return nextPolicy.Send(request, context);
+    }
```

Comment:
What's the point of this check? Comment please.

Reply:
When recodId is not set, it means the proxy policy has nothing to do.  This is known behavior across languages.  But sure, I can add a comment :)